scripts/coverage.sh

```diff
@@ -1,136 +1,69 @@
+rm -rf "$here/../target" "$here/../coverage"
+find "$here/.." -type f -name '*.prof*' -exec rm {} +
+find "$here/.." -type f -name '*lcov*' -exec rm {} +
 
-: "${CI_COMMIT:=local}"
-reportName="lcov-${CI_COMMIT:0:9}"
+export RUSTFLAGS="-C instrument-coverage $RUSTFLAGS"
+export LLVM_PROFILE_FILE="default-%p-%m.profraw"
 
 if [[ -z $1 ]]; then
-  packages=(--lib --all --exclude solana-local-cluster)
+  PACKAGES=(--lib --all --exclude solana-local-cluster)
 else
-  packages=("$@")
-fi
-
-coverageFlags=()
-coverageFlags+=(-Zprofile)               # Enable coverage
-coverageFlags+=("-Aincomplete_features") # Suppress warnings due to frozen abi, which is harmless for it
-if [[ $(uname) != Darwin ]]; then        # macOS skipped due to https://github.com/rust-lang/rust/issues/63047
-  coverageFlags+=("-Clink-dead-code")    # Dead code should appear red in the report
-fi
-coverageFlags+=("-Ccodegen-units=1")     # Disable code generation parallelism which is unsupported under -Zprofile (see [rustc issue #51705]).
-coverageFlags+=("-Cinline-threshold=0")  # Disable inlining, which complicates control flow.
-coverageFlags+=("-Copt-level=0")
-coverageFlags+=("-Coverflow-checks=off") # Disable overflow checks, which create unnecessary branches.
-
-export RUSTFLAGS="${coverageFlags[*]} $RUSTFLAGS"
-export CARGO_INCREMENTAL=0
-export RUST_BACKTRACE=1
-export RUST_MIN_STACK=8388608
-
-echo "--- remove old coverage results"
-if [[ -d target/cov ]]; then
-  find target/cov -type f -name '*.gcda' -delete
+  PACKAGES=("$@")
 fi
-rm -rf target/cov/$reportName
-mkdir -p target/cov
-
-# Mark the base time for a clean room dir
-touch target/cov/before-test
-
-# Force rebuild of possibly-cached proc macro crates and build.rs because
-# we always want stable coverage for them
-# Don't support odd file names in our repo ever
-if [[ -n $CI || -z $1 ]]; then
-  # shellcheck disable=SC2046
-  touch \
-    $(git ls-files :**/build.rs) \
-    $(git grep -l "proc-macro.*true" :**/Cargo.toml | sed 's|Cargo.toml|src/lib.rs|')
-fi
-
-#shellcheck source=ci/common/limit-threads.sh
-source ci/common/limit-threads.sh
-
-_ "$cargo" nightly test --jobs "$JOBS" --target-dir target/cov --no-run "${packages[@]}"
 
 TEST_ARGS=(
-  --nocapture
   --skip shred::merkle::test::test_make_shreds_from_data::
   --skip shred::merkle::test::test_make_shreds_from_data_rand::
   --skip shred::merkle::test::test_recover_merkle_shreds::
 )
 
 # most verbose log level (trace) is enabled for all solana code to make log!
 # macro code green always
-if RUST_LOG=solana=trace _ ci/intercept.sh "$cargo" nightly test --jobs "$JOBS" --target-dir target/cov "${packages[@]}" -- "${TEST_ARGS[@]}" ; then
-  test_status=0
-else
-  test_status=$?
-  echo "Failed: $test_status"
-  echo "^^^ +++"
-  if [[ -n $CI ]]; then
-    exit $test_status
-  fi
-fi
-touch target/cov/after-test
+RUST_LOG="solana=trace agave=trace $RUST_LOG" ci/intercept.sh cargo +"$rust_nightly" test "${PACKAGES[@]}" -- "${TEST_ARGS[@]}"
```

Comment:
oops, my bad: the correct syntax is this...: `solana=trace,agave=trace,$RUST_LOG`

Reply:
ref: https://docs.rs/env_logger/latest/env_logger/#enabling-logging

Reply:
thank you! [15f8544](https://github.com/anza-xyz/agave/pull/1260/commits/15f854441f38abc678ff24d98cbbe9002616a403)

Reply:
btw, what's the main reason that we want to keep trace level log in CI? I think if something happen, we will still need to try to reproduce it in local. we can turn on the log at that time. I think most of trace log are too trivial and it spams our space.

Reply:
and there is a bad news, when we turn on the correct log level, coverage test failed 🫠

https://buildkite.com/anza/agave/builds/4386#_ failed 4 times
https://buildkite.com/anza/agave/builds/4390#018f6268-f845-48e8-9105-8f39b4002d29 failed 1 time

all of them reported:
```
---- banking_stage::consumer::tests::test_write_persist_loaded_addresses stdout ----
thread 'banking_stage::consumer::tests::test_write_persist_loaded_addresses' panicked at runtime/src/bank.rs:1816:39:
attempt to multiply with overflow
```
I can also reproduce it in local easily.

could we turn off the trace level log if we don't have a strongly needed? or maybe we ignore this one in this PR and restore in another PR?

Reply:
ohh, thanks for trying... okay, I'll restore it later.